tiledb/sm/consolidator/commits_consolidator.cc

```diff
@@ -79,17 +79,17 @@ Status CommitsConsolidator::consolidate(
   }
 
   // Get the array uri to consolidate from the array directory.
-  ArrayDirectory array_dir;
-  try {
-    array_dir = ArrayDirectory(
-        storage_manager_->vfs(),
-        storage_manager_->compute_tp(),
-        URI(array_name),
-        0,
-        utils::time::timestamp_now_ms(),
-        ArrayDirectoryMode::COMMITS);
-  } catch (const std::logic_error& le) {
-    return LOG_STATUS(Status_ArrayDirectoryError(le.what()));
+  auto array_dir = ArrayDirectory(
```

Comment:
The try/catch was there because if the array doesn't exists, we'll get an exception... I assume that's ok now with our new exception model?

Reply:
If so could I ask for a favor and search for "array_dir = ArrayDirectory(" in the codebase and remove the same code in all other places?

Reply:
@KiterLuc done. Also added tests for sparse and the other relevant modes.